### PR TITLE
[Addition of Custom User class specific to polls app] & [Addition of UserViewSet in API]

### PR DESCRIPTION
**Pull request by Dasher83**
Migrations where changed in order to introduce a custom User class specific to the polls app, instead of using the User model global to Django (default).

In addition, some basic views where added to the API app to manage users through them. Authorization and Authentication is still not included. Automated tests would be included after PR is accepted.

### CODE REVIEW

File: mysite/api/views.py [hunk 1]
```diff
@@ -15,3 +15,13 @@ class QuestionViewSet(viewsets.ModelViewSet):
...
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    serializer_class = UserRetrieveSerializer
+    queryset = User.objects.all()
+    serializer_action_classes = {
+        'list': UserListSerializer,
+        'create': UserCreateSerializer,
+        'update': UserUpdateSerializer
+    }
```
Comment (joaquinco) [inline mysite/api/views.py]: Is this being used? I don't think DRF supports it

Comment (Dasher83) [reply mysite/api/views.py]: It is being used exclusively in this new UserViewSet. It seems to be working fine. I'll write tests on it to check it is being properly supported.
I used the following article as reference.
[https://medium.com/aubergine-solutions/decide-serializer-class-dynamically-based-on-viewset-actions-in-django-rest-framework-drf-fb6bb1246af2](url)

**Review by dcousin24 — APPROVED** [review]

File: mysite/api/views.py (hunk 1 above)
Comment (dcousin24) [reply mysite/api/views.py]: It works but if you return it in the serializer in the get_serializer_class method of the viewset I think. By it's own it won't make anything

Comment (Dasher83) [reply mysite/api/views.py]: Oh, now I see. Yes, even the very article I use as an argument explains what you both are saying. I did not override the get_serializer_class method. I'll get on it ASAP.

**Review by MBerguer — APPROVED** [review]
The code is great!
For future iterations; there are some things that we could improve, like using Inheritance for **create_superuser** and **create_user**. I will leave it up to you if you want to do some research about that, but there is no need to do it for this PR. 

Also, I noticed some formatting issues through the code, which IDE are you using? I strongly recommend that you use Pycharm, it has some cool features that will save you some time while coding.

**Review by joaquinco — APPROVED** [review]

**Review by dcousin24 — APPROVED** [review]

**Pull request merged**